### PR TITLE
AWS EBS CSI Driver: Add `node_affinity` and `tolerations`.

### PR DESCRIPTION
**Pull request by surajssd**
- Update component from `0.7.0` to `0.9.0`.
    
- Remove the component specific change of allowing a SC to be default
  using annotation. Use the helm chart's default mechanism to provide
  storage class.
    
- This PR adds two params to the component viz. `node_affinity` and
  `tolerations`.
    
- Add unit tests for the same.

- Add documentation to configuration reference docs.

Fixes: #1385

### CODE REVIEW

File: pkg/components/aws-ebs-csi-driver/component.go [hunk 1]
```diff
@@ -39,6 +39,15 @@ enableVolumeScheduling: {{ .EnableVolumeScheduling }}
...
 enableVolumeSnapshot: {{ .EnableVolumeSnapshot }}
+
+storageClasses:
+- name: ebs-sc
+  {{ if .EnableDefaultStorageClass }}
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+  {{ end }}
+  volumeBindingMode: WaitForFirstConsumer
+  reclaimPolicy: Retain
```
Comment (invidian) [inline pkg/components/aws-ebs-csi-driver/component.go:50]: Why Retain as a default? This change has not been mentioned in commit message and it's a functional change.

File: pkg/components/aws-ebs-csi-driver/component.go [hunk 2]
```diff
@@ -48,14 +48,34 @@ storageClasses:
...
-	EnableVolumeResizing      bool `hcl:"enable_volume_resizing,optional"`
-	EnableVolumeSnapshot      bool `hcl:"enable_volume_snapshot,optional"`
+	EnableDefaultStorageClass bool              `hcl:"enable_default_storage_class,optional"`
+	EnableVolumeScheduling    bool              `hcl:"enable_volume_scheduling,optional"`
+	EnableVolumeResizing      bool              `hcl:"enable_volume_resizing,optional"`
+	EnableVolumeSnapshot      bool              `hcl:"enable_volume_snapshot,optional"`
+	Tolerations               []util.Toleration `hcl:"tolerations,block"`
+	TolerationsRaw            string
+	NodeAffinity              []util.NodeAffinity `hcl:"node_affinity,block"`
+	NodeAffinityRaw           string
```
Comment (invidian) [inline pkg/components/aws-ebs-csi-driver/component.go:78]: Can we keep not-user-accessible fields at the bottom?

File: pkg/components/aws-ebs-csi-driver/component_test.go [hunk 3]
```diff
@@ -137,3 +139,133 @@ func TestStorageClassDisabled(t *testing.T) {
...
+			name: "affinity_csi_controller",
+			inputConfig: `component "aws-ebs-csi-driver" {
+							node_affinity {
+								key      = "lokomotive.io/role"
+								operator = "In"
+								values   = ["storage"]
+							}
+						}`,
+			expectedManifestName: "aws-ebs-csi-driver/templates/controller.yaml",
+			jsonPath:             "{.spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key}",
```
Comment (invidian) [inline pkg/components/aws-ebs-csi-driver/component_test.go:225]: You can break the string with `+` operator, so it's not that long. Then we can keep `lll` linter.

File: pkg/components/aws-ebs-csi-driver/component.go (hunk 1 above)
Comment (surajssd) [reply pkg/components/aws-ebs-csi-driver/component.go:50]: Added

File: pkg/components/aws-ebs-csi-driver/component_test.go [hunk 4]
```diff
@@ -137,3 +139,134 @@ func TestStorageClassDisabled(t *testing.T) {
...
+		{
+			name: "tolerations_csi_controller",
+			inputConfig: `component "aws-ebs-csi-driver" {
+							tolerations {
+								key      = "lokomotive.io"
+								operator = "Equal"
+								value    = "awesome"
+								effect   = "NoSchedule"
+							}
+						}`,
```
Comment (invidian) [inline pkg/components/aws-ebs-csi-driver/component_test.go:188]: Indentation is interesting in those test cases :smile: But I'm not sure what would be the right way to format it.

**Review by invidian — APPROVED** [review]
I'm not sure if `Retain` is a good default policy, but time will tell. LGTM

**Review by invidian — APPROVED** [review]

**Pull request merged**